CMakeLists.txt

```
@@ -1,75 +1,73 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.8)
 project(mpu6050_driver)
 
-add_compile_options(-std=c++11)
-# add_definitions(-Wall -Werror)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-set(CATKIN_DEPS 
-  # diagnostic_msgs
-  roscpp
-  sensor_msgs
-  i2c_device_ros
-)
-
-find_package(catkin REQUIRED COMPONENTS ${CATKIN_DEPS})
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
-find_package(cmake_modules REQUIRED)
-
-catkin_package(
- INCLUDE_DIRS include
- CATKIN_DEPENDS ${CATKIN_DEPS}
- DEPENDS Eigen
-)
 
 include_directories(
   include
-  ${catkin_INCLUDE_DIRS}
   ${Eigen3_INCLUDE_DIRS}
 )
 
-set(MPU6050_NODE_LIB mpu6050_node_lib)
-add_library(${MPU6050_NODE_LIB}
+# Define library target
+add_library(mpu6050_node_lib
   src/mpu6050_node.cpp
   src/mpu6050.cpp
+  src/i2c_device.cpp
 )
+ament_target_dependencies(mpu6050_node_lib rclcpp sensor_msgs)
 
-add_executable(mpu6050_node 
-  src/main.cpp
-)
+# Link against ROS 2 libraries
+# target_link_libraries(mpu6050_node_lib
+#   ${rclcpp_LIBRARIES}
+# )
 
-target_link_libraries(mpu6050_node
-  ${catkin_LIBRARIES}
-  ${MPU6050_NODE_LIB}
-)
+# Define executable targets
+add_executable(mpu6050_node src/main.cpp)
+ament_target_dependencies(mpu6050_node rclcpp sensor_msgs)
+target_link_libraries(mpu6050_node mpu6050_node_lib)
```

```suggestion
add_executable(mpu6050_node
  src/main.cpp
)

ament_target_dependencies(mpu6050_node
  rclcpp
  sensor_msgs
)

target_link_libraries(mpu6050_node
  mpu6050_node_lib
)
```

Nitpick to comply with what we do in our internal repositories. Please apply this to the following lines, too.